spec/models/spell_finder_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe 'spell index can be filtered' do
+  let(:spell_name) { 'Level 1 Paladin Spell' }
+  let(:paladin_level) { 1 }
+  let!(:paladin_spell) do
+    FactoryGirl.create(
+      :spell,
+      name: spell_name,
+      class_requirements: {
+        alchemist: nil,
+        antipaladin: nil,
+        bard: nil,
+        bloodrager: nil,
+        cleric: 1,
+        druid: nil,
+        hunter: nil,
+        inquisitor: 1,
+        investigator: nil,
+        magus: nil,
+        medium: nil,
+        mesmerist: nil,
+        occultist: nil,
+        oracle: 1,
+        paladin: paladin_level,
+        psychic: nil,
+        ranger: nil,
+        shaman: 1,
+        skald: nil,
+        sorcerer: nil,
+        spiritualist: nil,
+        summoner: nil,
+        witch: nil,
+        wizard: nil,
+      },
+    )
+  end
+
+  let(:params) { { class_filter: :paladin } }
+  let(:spell_finder) { SpellFinder.new(params) }
+
+  it 'index can be filtered to show all spells of a class' do
+    expect(spell_finder.perform).to match_array([paladin_spell])
+  end
+
+  context 'with a specific level' do
+    let(:paladin_level) { 2 }
+    let(:params) { { class_filter: :paladin, level: 2 } }
+    let(:out_of_scope_spell_level) { 1 }
```

Comment:
What do the `{ 1 }` and `{ 2 }` signify or do?

Reply:
Ah these are the class levels but I forgot to add them to the params here